Sources/APIProvider.swift

```diff
@@ -135,28 +122,38 @@ extension DataRequest {
     ///   - type: The type to map to.
     ///   - completion: The result of the mapping. An error will be returned if mapping fails.
     @discardableResult
-    func mapResponseTo<T: Decodable>(_ type: T.Type, decoder: JSONDecoder, completion: @escaping RequestCompletionHandler<T>) -> Self {
-        return responseData(queue: DispatchQueue.global(qos: .background)) { (response) in
-            if let error = response.error {
+    func dataResponse(decoder: JSONDecoder, completion: @escaping RequestCompletionHandler<Data?>) -> Self {
+        return validate(statusCode: 200..<300).responseData(queue: DispatchQueue.global(qos: .background)) {
+            if let error = $0.error {
                 // Try to parse api error
-                guard let data = response.data, let apiError = try? decoder.decode(ErrorResponse.self, from: data) else {
+                guard let data = $0.data, let apiError = try? decoder.decode(ErrorResponse.self, from: data) else {
                     completion(Result.failure(error))
                     return
                 }
                 completion(Result.failure(apiError))
             } else {
+                completion(Result.success($0.data))
+            }
+        }
+    }
+    
+    /// Maps the response to the given JSONDecodable data type.
+    ///
+    /// - Parameters:
+    ///   - type: The type to map to.
+    ///   - completion: The result of the mapping. An error will be returned if mapping fails.
+    @discardableResult
+    func mapResponseTo<T: Decodable>(_ type: T.Type, decoder: JSONDecoder, completion: @escaping RequestCompletionHandler<T>) -> Self {
+        return dataResponse(decoder: decoder) {
+            let result = $0.flatMap({ data -> T in
```

Comment:
same, I would keep response here for readability

Reply:
Also done